Firebend.AutoCrud.Mongo/Abstractions/Client/Indexing/MongoIndexComparisonService.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Generic;
+using System.Linq;
+using Firebend.AutoCrud.Mongo.Interfaces;
+using Firebend.JsonPatch.Extensions;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Driver;
+
+namespace Firebend.AutoCrud.Mongo.Abstractions.Client.Indexing
+{
+    public class MongoIndexComparisonService : IMongoIndexComparisonService
+    {
+        public bool DoesIndexMatch<TEntity>(IMongoCollection<TEntity> collection, BsonDocument existingIndexBson, CreateIndexModel<TEntity> definition)
+        {
+            if (existingIndexBson == null)
+            {
+                return false;
+            }
+
+            var doesNameMatch = existingIndexBson["name"].AsString.EqualsIgnoreCaseAndWhitespace(definition.Options.Name);
+
+            if (!doesNameMatch)
+            {
+                return false;
+            }
+
+            var existingKeys = existingIndexBson["key"].AsBsonDocument;
+            var definitionKeys = definition.Keys.Render(collection.DocumentSerializer, new BsonSerializerRegistry());
+            var doKeysMatch = existingKeys == definitionKeys;
+            var isFullText = definitionKeys.Contains("$**");
+
+            if (!isFullText)
+            {
+                return doKeysMatch;
+            }
+
+            if (existingIndexBson.Contains("weights"))
+            {
+                try
+                {
+                    var weightBson = existingIndexBson["weights"].AsBsonDocument;
+                    return weightBson.Contains("$**");
+                }
+                catch
+                {
```

Comment:
Did you mean to leave this here?

Reply:
no sir I'll clean it up shortly

Reply:
fixed up and ready to review. i also included a commit to validate payloads do not modify custom fields. they should use the custom fields endpoint for that.